kubernetes/product-deployment/main.jenkins

```diff
@@ -0,0 +1,286 @@
+                                name: 'db_engine',
+                                defaultValue: 'mysql',
+                                description: 'Database engine',
+                                trim: true
+                            ),
+                            string(
+                                name: 'db_version',
+                                defaultValue: '8.0.28',
+                                description: 'Database engine version',
+                                trim: true
+                            ),
+                            string(
+                                name: 'db_instance_class',
+                                defaultValue: 'db.t3.small',
+                                description: 'Database engine instance class',
+                                trim: true
+                            ),
+                            separator(name: "KUBERNETES_ENVIRONMENT", sectionHeader: "Kubernetes Details",
+                                separatorStyle: "border-width: 0",
+                                sectionHeaderStyle: """
+                                    background-color: #7ea6d3;
+                                    text-align: center;
+                                    padding: 4px;
+                                    color: #343434;
+                                    font-size: 22px;
+                                    font-weight: normal;
+                                    text-transform: uppercase;
+                                    font-family: 'Orienta', sans-serif;
+                                    letter-spacing: 1px;
+                                    font-style: italic;
+                                    margin-bottom: 25px;
+                                """),           
+                            string(
+                                name: 'kubernetes_repo_url',
+                                defaultValue: 'https://github.com/Tharsanan1/kubernetes-apim',
+                                description: "Product's kubernetes repository URL.",
+                                trim: true
+                            ),
+                            string(
+                                name: 'kubernetes_repo_branch',
+                                defaultValue: '4.1.x',
+                                description: 'Branch name of the kubernetes repository.',
+                                trim: true
+                            ),
+                            string(
+                                name: 'path_to_helm_folder',
+                                defaultValue: 'advanced/am-pattern-4',
+                                description: 'Relative path of the helm folder in your helm repository.',
+                                trim: true
+                            ),
+                            string(
+                                name: 'kubernetes_namespace',
+                                defaultValue: 'default',
```

Comment:
shall we have a separate namespace for the deployment rather than using the default? Maybe a namespace with the product name and it's version?

Reply:
Changed the default value to apim-4-1-0